### DISCUSSION TITLE
Grafana memory leak?

### DISCUSSION THREAD

**Discussion by atb2215** [apache/iotdb#10127] [Q&A] [answered]
Running several dashboards with IotDB version 1.1.1-SNAPSHOT (Build: c9b4dc2-dev) and Grafana v9.5.2.  
There seems to be a memory leak.  Running "show queries" gives hundreds of apparently unresolved queries 
+------------------------+-----------------------+----------+-----------+
|                    Time|                                                      QueryId|DataNodeId|ElapsedTime|
|2023-06-12T01:48:39.177Z|20230612_014839_00000_1|                   1|  12055.599|
|2023-06-12T01:48:39.178Z|20230612_014839_00001_1|                   1|  12055.598|
|2023-06-12T01:48:39.186Z|20230612_014839_00002_1|                  1|   12055.59 |
....
Running 'KILL ALL QUERIES' and 'CLEAR CACHE' seems to have no impact and query_timeout_threshold is set to 10000ms.  

Is this a bug or am I misunderstanding how "KILL QUERY" is intended to work?

**Comment by JackieTien97** [comment, marked as answer]
Can you give the full result set of `show queries`, what's exactly the running queries?
`KILL ALL QUERIES` will kill the running queries, but you should know that
1. It's an async opreation and you can take it as a kill signal which means that the actual killing operation will run in the background.
2. V1.1.2 IoTDB has fixed some memory dead lock bug, you can upgrade to it.

**Reply by atb2215** [reply]
Thanks @JackieTien97 , upgrading to 1.1.2 seems to have fixed it!
